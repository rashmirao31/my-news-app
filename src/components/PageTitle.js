import React from 'react';
import { Header } from 'semantic-ui-react';

class PageTitle extends React.Component {

  render() {
    return (
      <div data-testid='page-title'>
        <Header as='h1'>{this.props.selectedCountry} {window.defaultAppSettings.pageTitleSuffixLabel}</Header>
      </div>
    );
  };
  
}

export default PageTitle