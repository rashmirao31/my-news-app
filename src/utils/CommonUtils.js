var CommonUtils = {

	getConfigList: function() {
		let newsApiUrl = "https://newsapi.org/v2/";
		return {
			"newsApiUrl": newsApiUrl,
			"newsApiTopHeadlinesUrl": newsApiUrl + "top-headlines",
			"newsApiEverythingUrl": newsApiUrl + "everything",
			"apiKey": "687cf8fdc1d7435eb10214b70c78788d",
			"countryKey": "gb",
			"countryName": "United Kingdom",
			"category": "general",
			"timezoneKey": "Europe/London",
			"countryOptions": [
			  { key: 'gb', value: 'gb', flag: 'gb', text: 'United Kingdom' },
			  { key: 'us', value: 'us', flag: 'us', text: 'USA' },
			  { key: 'in', value: 'in', flag: 'in', text: 'India' }
			],
			"countryMenuTitle": "Country:",
			"countryMenuPlaceholder": "Select Country",
			"headerTimeLabel": "Updated at: ",
			"pageTitleSuffixLabel": "News",
			"searchInputPlaceholder": "Search...",
			"searchButtonText": "Search",
			"topStoriesText": "Top Stories",
			"searchResultHeaderText": "Search Results For",
			"loadingText": "Please wait...",
			"noDataText": "No results found",
			"readMoreText": "Read more",
			"errorScenarioText": "Something went wrong. Please try again after sometime."
		};
	},


};

export default CommonUtils;