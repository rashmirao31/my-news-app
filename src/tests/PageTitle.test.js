import React from 'react';
import ReactDOM from 'react-dom';
import CommonUtils from './../utils/CommonUtils';
import PageTitle from './../components/PageTitle.js';

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

beforeEach(() => {
  window.defaultAppSettings = CommonUtils.getConfigList();
  return;
});

afterEach(cleanup);

it('renders without crashing', () => {
 	const div = document.createElement('div');
	ReactDOM.render(<PageTitle />, div);
});

it('matches snapshot', () => {
 	const tree = renderer.create(
    	<PageTitle selectedCountry="India"/>
	).toJSON();
 	expect(tree).toMatchSnapshot();
});

it('renders the title for the country selected', () => {
 	const selectCountry = "India";
 	const { getByTestId } = render(
    	<PageTitle selectedCountry={selectCountry}/>
	);
	expect(getByTestId("page-title")).toHaveTextContent(selectCountry + " " + window.defaultAppSettings.pageTitleSuffixLabel);
});