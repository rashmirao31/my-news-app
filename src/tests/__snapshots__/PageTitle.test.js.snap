// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="page-title"
>
  <h1
    className="ui header"
  >
    India
     
    News
  </h1>
</div>
`;
